phpdotnet/phd/Options/Parser.php

```diff
@@ -4,23 +4,30 @@
 class Options_Parser
 {
 
-    private $defaultHandler;
-    private $packageHandlers = array();
+    private Options_Interface $defaultHandler;
+    private array $packageHandlers = [];
 
-    public function __construct() {
-        $this->defaultHandler = new Options_Handler();
-        $this->packageHandlers = $this->loadPackageHandlers();
+    /**
+     * @param array<?Options_Interface> $packageHandlers
+     */
+    public function __construct(Options_Interface $defaultHandler, ?array $packageHandlers = []) {
```

Comment:
Possible idea for latter, make ` $packageHandlers` a variadic argument of type `Options_Interface` ?